app/models/ride.rb

```diff
@@ -6,4 +6,24 @@ class Ride < ActiveRecord::Base
   validates :seat_count, presence: true
   validates :date, presence: true
   validates :departure_time, presence: true
+
+  def full?
+    passengers.count >= seat_count
+  end
+
+  def has_passenger?(user)
+    passengers.any? { |passenger| passenger.user_id == user.id }
+  end
+
+  def add_passenger(passenger)
+    Passenger.create ride: self, user: passenger
```

Comment:
you could use `passengers.create user: passenger`. Rails will automatically include the ride because of the relationship.